src/shared/tests.h

```diff
@@ -200,3 +200,42 @@ static inline int run_test_table(void) {
         DEFINE_TEST_MAIN_FULL(log_level, intro, NULL)
 #define DEFINE_TEST_MAIN(log_level)                     \
         DEFINE_TEST_MAIN_FULL(log_level, NULL, NULL)
+
+#define ASSERT_OK(expr)                                                                                  \
+        ({                                                                                               \
+                int _result = (expr);                                                                    \
+                if (_result < 0) {                                                                       \
+                        log_error_errno("Assertion failed: %s (result: %d, error: %m)", #expr, _result); \
```

Comment:
Drop `result: %d` from this one. We already show the result via the error in a more readable form.